mmv1/third_party/terraform/services/storage/resource_storage_anywhere_cache_test.go

```
@@ -0,0 +1,94 @@
+			{
+				Config: testAccStorageAnywhereCache_full(context),
+			},
+			{
+				ResourceName:            "google_storage_anywhere_cache.cache",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"bucket"},
+			},
+			{
+				Config: testAccStorageAnywhereCache_update(context),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_storage_anywhere_cache.cache", plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+			{
+				ResourceName:            "google_storage_anywhere_cache.cache",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"bucket"},
+			},
+		},
+	})
+}
+
+func testAccStorageAnywhereCache_full(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_storage_bucket" "bucket" {
+  name                        = "tf-test-bucket-name%{random_suffix}"
+  location                    = "US"
+}
+
+resource "time_sleep" "destroy_wait_5000_seconds" {
+  depends_on = [google_storage_bucket.bucket]
+  destroy_duration = "5000s"
+}
+
+resource "google_storage_anywhere_cache" "cache" {
+  bucket = google_storage_bucket.bucket.name
+  zone = "us-central1-f"
+  ttl = "3601s"
+  depends_on = [time_sleep.destroy_wait_5000_seconds]
+}
+`, context)
+}
+
+func testAccStorageAnywhereCache_update(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_storage_bucket" "bucket" {
+  name                        = "tf-test-bucket-name%{random_suffix}"
+  location                    = "US"
+}
+
+resource "time_sleep" "destroy_wait_5000_seconds" {
+  depends_on = [google_storage_bucket.bucket]
+  destroy_duration = "5000s"
+}
+
+resource "google_storage_anywhere_cache" "cache" {
+  bucket = google_storage_bucket.bucket.name
+  zone = "us-central1-f"
+  admission_policy = "admit-on-second-miss"
+  ttl = "3620s"
```

Test failure indicates that the API is silently ignoring this value on update.

---

Possible explanations include: something wrong with the update mask? eventual consistency on update? field isn't actually updatable?

---

It looks like, while doing an update - some of the fields do get updated immediately , while on the other hand rest of fields take some time to update. For above example - `admission_policy` updates instantly, and `ttl` takes `30-40`min of time for which there's an LRO.  There's a field `pending_update` to check if an update is pending on resource. So `ttl` wouldn't be set instantly in the metadata, instead `cache` would be set.

One solutions is to waiting on update LRO, But i'm suspecting there might be a better solution.

---

I'd recommend waiting for the LRO if that will guarantee that the updates are completed, unless there's a specific need not to.

---

The other option would be to have a separate `desired_ttl` vs `ttl` field so that the user's goal and the actual current value are separate.

---

Looking back, I realized I asked for not waiting for the LRO - sorry about that! I thought it was just for TTL rollout, not the field value changing.

---

Done